### PR TITLE
fix: update code to newer method signatures

### PR DESCRIPTION
**Pull request by matskiv**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
Resolves the compilation errors.

**Please provide a short message that should be published in the vcluster release notes**
N/A

### CODE REVIEW

File: pkg/controllers/generic/patcher.go [hunk 1]
```diff
@@ -56,7 +57,8 @@ func (s *patcher) ApplyPatches(ctx context.Context, fromObj, toObj client.Object
 		// always apply status if it's there
 		if hasAfterStatus {
 			s.log.Infof("Apply status of %s during patching", toObjCopied.GetName())
-			err = s.toClient.Status().Patch(ctx, toObjCopied.DeepCopy(), client.Apply, client.ForceOwnership, client.FieldOwner(fieldManager))
+			o := &client.SubResourcePatchOptions{PatchOptions: client.PatchOptions{FieldManager: fieldManager, Force: pointer.Bool(true)}}
```
Comment (matskiv) [inline pkg/controllers/generic/patcher.go:60]: Based on:
https://github.com/kubernetes-sigs/controller-runtime/blob/4a538741e9aea74df6caf4670b05520b8adb6ea2/pkg/client/options.go#L786-L789
and
https://github.com/kubernetes-sigs/controller-runtime/blob/4a538741e9aea74df6caf4670b05520b8adb6ea2/pkg/client/options.go#L143-L145

File: pkg/server/server.go [hunk 2]
```diff
@@ -313,7 +313,7 @@ func initAdmission(ctx context.Context, vConfig *rest.Config) (admission.Interfa
 		&emptyConfigProvider{},
 		admission.PluginInitializers{
 			webhookinit.NewPluginInitializer(authInfoResolverWrapper, serviceResolver),
-			initializer.New(vClient, kubeInformerFactory, nil, nil, nil),
+			initializer.New(vClient, nil, kubeInformerFactory, nil, nil, nil),
```
Comment (matskiv) [inline pkg/server/server.go:316]: The second parameter is for now used only in the "[celAdmissionPlugin](https://github.com/kubernetes/kubernetes/blob/97ab147537f6999249c241e0896245bd961c32ae/staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/admission.go#LL79C27-L79C27)", but we only register the "mutating" and "validating" plugins.

Comment (neogopher) [reply pkg/server/server.go:316]: ok

File: pkg/controllers/generic/patcher.go (hunk 1 above)
Comment (neogopher) [reply pkg/controllers/generic/patcher.go:60]: Yes. I understood this one now.

**Pull request merged**